### PR TITLE
Fix #1920: handle raw Iterator types when building iterator method access tree

### LINKED ISSUE #1920: assertion failure in CFGBuilder for erased Iterator type
This is with CF 2.4.0.  Input `Test.java`:
```java
import java.util.*;
class Test {
  static class Foo implements Iterable {
    public Iterator iterator() {
      return new Iterator() {
        @Override
        public boolean hasNext() {
          return false;
        }

        @Override
        public Object next() {
          throw new NoSuchElementException();
        }
      };
    }
  }
  static void testErasedIterator(Foo foo) {
    for (Object x : foo) {
      x.hashCode();
    }
  }
}
```
Output:
```
$ javac -AprintErrorStack -processor nullness Test.java
error: SourceChecker.typeProcess: unexpected Throwable (AssertionError) while processing Test.java; message: expected exactly one type argument for Iterator
  Compilation unit: Test.java
  Exception: java.lang.AssertionError: expected exactly one type argument for Iterator; Stack trace: org.checkerframework.javacutil.trees.TreeBuilder.buildIteratorMethodAccess(TreeBuilder.java:104)
  org.checkerframework.dataflow.cfg.CFGBuilder$CFGTranslationPhaseOne.visitEnhancedForLoop(CFGBuilder.java:3589)
  org.checkerframework.dataflow.cfg.CFGBuilder$CFGTranslationPhaseOne.visitEnhancedForLoop(CFGBuilder.java:1489)
  com.sun.tools.javac.tree.JCTree$JCEnhancedForLoop.accept(JCTree.java:1043)
  com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:68)
  org.checkerframework.dataflow.cfg.CFGBuilder$CFGTranslationPhaseOne.visitBlock(CFGBuilder.java:3303)
  org.checkerframework.dataflow.cfg.CFGBuilder$CFGTranslationPhaseOne.visitBlock(CFGBuilder.java:1489)
  com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:918)
  com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:50)
  org.checkerframework.dataflow.cfg.CFGBuilder$CFGTranslationPhaseOne.process(CFGBuilder.java:1607)
  org.checkerframework.dataflow.cfg.CFGBuilder$CFGTranslationPhaseOne.process(CFGBuilder.java:1654)
  org.checkerframework.framework.flow.CFCFGBuilder.run(CFCFGBuilder.java:63)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.analyze(GenericAnnotatedTypeFactory.java:1200)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.analyze(GenericAnnotatedTypeFactory.java:1177)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.performFlowAnalysis(GenericAnnotatedTypeFactory.java:1107)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.checkAndPerformFlowAnalysis(GenericAnnotatedTypeFactory.java:1494)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.preProcessClassTree(GenericAnnotatedTypeFactory.java:248)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:291)
  org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:162)
  com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:720)
  com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:50)
  org.checkerframework.framework.source.SourceVisitor.visit(SourceVisitor.java:66)
  org.checkerframework.framework.source.SourceChecker.typeProcess(SourceChecker.java:967)
  org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:502)
  org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:495)
  org.checkerframework.javacutil.AbstractTypeProcessor$AttributionTaskListener.finished(AbstractTypeProcessor.java:182)
  com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:681)
  com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:111)
  com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1342)
  com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1296)
  com.sun.tools.javac.main.JavaCompiler.compile2(JavaCompiler.java:901)
  com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:860)
  com.sun.tools.javac.main.Main.compile(Main.java:523)
  com.sun.tools.javac.main.Main.compile(Main.java:381)
  com.sun.tools.javac.main.Main.compile(Main.java:370)
  com.sun.tools.javac.main.Main.compile(Main.java:361)
  com.sun.tools.javac.Main.compile(Main.java:56)
  com.sun.tools.javac.Main.main(Main.java:42)
```

### PR DESCRIPTION
**Pull request by msridhar**

### CODE REVIEW

File: javacutil/src/org/checkerframework/javacutil/trees/TreeBuilder.java [hunk 1]
```diff
@@ -101,17 +101,19 @@ public MemberSelectTree buildIteratorMethodAccess(ExpressionTree iterableExpr) {
...
+
+        if (numIterTypeArgs == 1) {
+            TypeMirror elementType = iteratorType.getTypeArguments().get(0);
+            // Remove captured type from a wildcard.
+            if (elementType instanceof Type.CapturedType) {
+                elementType = ((Type.CapturedType) elementType).wildcard;
+
+                iteratorType =
+                        modelTypes.getDeclaredType(
+                                (TypeElement) modelTypes.asElement(iteratorType), elementType);
```
Comment (msridhar) [inline javacutil/src/org/checkerframework/javacutil/trees/TreeBuilder.java]: Not clear to me whether the subsequent `updatedMethodType` code is needed in the case of a raw Iterator type.  But seemed to me it should still work, so I didn't touch it to make this change as narrow as possible

**Comment by msridhar** [comment]
Is there any way to confirm that the test case I added was run by Travis?  I tried looking through the logs for the different jobs but couldn't find it

**Comment by mernst** [comment]
@msridhar Putting a file in `framework/tests/all-systems/` should just work, and I would expect it to.

A hacky way (but it works) is to create a new branch, make the test intentionally fail, and see that Travis fails as a result.

The individual tests are not printed by default by the testing framework.  You can use the `-Afilenames` command-line argument to print files as they are processed.

**Comment by smillst** [comment]
> Is there any way to confirm that the test case I added was run by Travis? I tried looking through the logs for the different jobs but couldn't find it.

No, Travis doesn't output which test cases are run.  You can check locally like this:
```
./gradlew ValueTest -Pemit.test.debug=true
```
Where `ValueTest` is the name of any Junit test class that uses all-systems.  But, like Mike said, put the test in framework/test/all-systems works.  Thanks!

**Comment by msridhar** [comment]
Thanks for merging! I ended up pushing another branch without the fix to
confirm that Travis was running the test.

**Pull request merged**